tedana/io.py

```diff
@@ -680,3 +682,43 @@ def split_ts(data, mmix, mask, comptable):
     resid = data - hikts
 
     return hikts, resid
+
+
+def prep_data_for_json(d) -> dict:
+    """Attempts to create a JSON serializable dictionary from a data dictionary
+
+    Parameters
+    ----------
+    d: dict
+        A dictionary that will be converted into something JSON serializable
+
+    Raises
+    ------
+    ValueError if it cannot force the dictionary to be serializable
+    TypeError if you do not supply a dict
+
+    Returns
+    -------
+    An attempt at JSON serializable data
+
+    Notes
+    -----
+    Use this to make something seriazable when writing JSON to disk.
```

Comment:
```suggestion
    Currently converts numpy arrays to lists and raises an error if there's
    another data structure that can be automatically written to JSON.
    Use this to make something serializable when writing JSON to disk.
```

Really minor suggested change to make clear this is really only for correcting one thing right now (accept or reject this suggestion). Otherwise LGTM.

Reply:
I like this suggestion, but I think it should say "another data structure that _cannot_ be automatically written to JSON."

Reply:
I think it's distinctly possible that more could come, and I'd rather not have to change the docstring and lead people to believe they should expand the docstring rather than just chucking another type into the if-block below.

Reply:
Like I said. I'm fine other way. The plus is, if I look at in a year, the note will help me remember why this function was needed. The minus is, as you note, it might over-promise future additions or over-specify. Resolve this review with or without making the change and I'll then approve.

Reply:
Latest commit modifies the language to make it clear that this function should be used to add more conversion cases without specifying them, that feels pretty split down the middle.